libs/libc/locale/lib_gettext.c

```diff
@@ -312,6 +312,7 @@ FAR char *dcngettext(FAR const char *domainname,
           return notrans;
         }
 
+      strncpy(mofile->path, path, PATH_MAX);
```

Comment:
strlcpy is better than stncpy since strncpy will set the whole unused space to zero.